File: php/src/Exceptions/TempMailException.php
<?php

declare(strict_types=1);

namespace TempMail\Exceptions;

class TempMailException extends \RuntimeException
{
    public function __construct(
        string $message = '',
        int $code = 0,
        ?\Throwable $previous = null,
        public readonly array $context = [],
    ) {
        parent::__construct($message, $code, $previous);
    }
}

File: java/src/main/java/com/tempmail/errors/TempMailException.java
package com.tempmail.errors;

public class TempMailException extends Exception {

    private final int statusCode;

    public TempMailException(String message) {
        super(message);
        this.statusCode = -1;
    }

    public TempMailException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = -1;
    }

    public TempMailException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public TempMailException(String message, int statusCode, Throwable cause) {
        super(message, cause);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}
